android/app/src/main/java/au/com/robin/sms/MainActivity.kt

```diff
@@ -5,32 +5,47 @@ import android.content.Intent
 import android.content.pm.PackageManager
 import android.os.Bundle
 import android.util.Log
+import android.widget.Button
+import android.widget.EditText
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.app.ActivityCompat
 import androidx.core.content.ContextCompat
-import au.com.robin.sms.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity() {
-    private lateinit var smsInterfaceBinding: ActivityMainBinding
-    private val PERMISSION_REQUEST_CODE = 1
+
+    /**
+     * Constant variables
+     */
+    companion object {
+        const val PERMISSION_REQUEST_CODE = 1
+        const val SIM_SLOT = "slot"
+        const val SIM_SLOT_ONE = 0 // SIM slot number 1 in a multi-sim phone.
+    }
+
+    /**
+     * Declare variables with lateinit that are `not-null` outside constructor.
+     */
+    private lateinit var btSend: Button
```

Comment:
does the activity class need to keep btSend, etPhineNo and etMessage as properties? I kinda wanna avoid lateinits if we can help ot